pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -398,6 +396,15 @@ public Pair<TableConfig, Schema> getCachedTableConfigAndSchema() {
     return _cachedTableConfigAndSchema;
   }
 
+  @Override
+  public void refreshCachedTableConfigAndSchema() {
```

Comment:
We need to invoke `TimestampIndexUtils.applyTimestampIndex()` which could modify `schema` before caching it. We can also consider just call `fetchIndexLoadingConfig()` without introducing this method